Provider/src/FirebirdSql.Data.FirebirdClient/Client/Managed/Version10/GdsDatabase.cs

```diff
@@ -378,31 +326,28 @@ protected void ProcessCreateResponse(GenericResponse response)
 
 		public virtual void DropDatabase()
 		{
-			lock (SyncObject)
+			try
 			{
-				try
-				{
-					XdrStream.Write(IscCodes.op_drop_database);
-					XdrStream.Write(_handle);
-					XdrStream.Flush();
+				XdrStream.Write(IscCodes.op_drop_database);
+				XdrStream.Write(_handle);
+				XdrStream.Flush();
 
-					ReadResponse();
+				ReadResponse();
 
-					_handle = 0;
-				}
-				catch (IOException ex)
-				{
-					throw IscException.ForErrorCode(IscCodes.isc_network_error, ex);
-				}
-				finally
+				_handle = 0;
+			}
+			catch (IOException ex)
+			{
+				throw IscException.ForErrorCode(IscCodes.isc_network_error, ex);
+			}
+			finally
+			{
+				try
 				{
-					try
-					{
-						CloseConnection();
-					}
-					catch
-					{ }
+					CloseConnection();
```

Comment:
There's a bunch of these exception-swallowing pieces across the code base and I don't know whether I like it. I tend to think the exception should be propagated. It's only few pieces where swallowing makes sense. See also my 3. in PR reply.